components/ledger/pkg/storage/sqlstorage/ledger/transactions.go

```diff
@@ -16,117 +16,165 @@ import (
 	"github.com/formancehq/stack/libs/go-libs/metadata"
 	"github.com/pkg/errors"
 	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/extra/bunbig"
 )
 
 const (
 	TransactionsTableName = "transactions"
 	PostingsTableName     = "postings"
 )
 
-// this regexp is used to distinguish between deprecated regex queries for
-// source, destination and account params and the new wildcard query
-// which allows segmented address pattern matching, e.g; "foo:bar:*"
-var addressQueryRegexp = regexp.MustCompile(`^(\w+|\*|\.\*)(:(\w+|\*|\.\*))*$`)
-
-type Transactions struct {
+type Transaction struct {
 	bun.BaseModel `bun:"transactions,alias:transactions"`
 
-	ID                uint64          `bun:"id,type:bigint,pk"`
-	Timestamp         core.Time       `bun:"timestamp,type:timestamptz"`
-	Reference         string          `bun:"reference,type:varchar,unique,nullzero"`
-	Postings          json.RawMessage `bun:"postings,type:jsonb"`
-	Metadata          json.RawMessage `bun:"metadata,type:jsonb,default:'{}'"`
-	PreCommitVolumes  json.RawMessage `bun:"pre_commit_volumes,type:jsonb"`
-	PostCommitVolumes json.RawMessage `bun:"post_commit_volumes,type:jsonb"`
+	ID        uint64            `bun:"id,type:bigint,pk"`
+	Timestamp core.Time         `bun:"timestamp,type:timestamptz"`
+	Reference string            `bun:"reference,type:varchar,unique,nullzero"`
+	Postings  []Posting         `bun:"rel:has-many,join:id=txid"`
+	Metadata  metadata.Metadata `bun:"metadata,type:jsonb,default:'{}'"`
+	//TODO(gfyrag): change to bytea
+	PreCommitVolumes  core.AccountsAssetsVolumes `bun:"pre_commit_volumes,type:bytea"`
+	PostCommitVolumes core.AccountsAssetsVolumes `bun:"post_commit_volumes,type:bytea"`
+}
+
+func (t Transaction) toCore() core.ExpandedTransaction {
+	postings := core.Postings{}
+	for _, p := range t.Postings {
+		postings = append(postings, p.toCore())
+	}
+	return core.ExpandedTransaction{
+		Transaction: core.Transaction{
+			TransactionData: core.TransactionData{
+				Postings:  postings,
+				Reference: t.Reference,
+				Metadata:  t.Metadata,
+				Timestamp: t.Timestamp,
+			},
+			ID: t.ID,
+		},
+		PreCommitVolumes:  t.PreCommitVolumes,
+		PostCommitVolumes: t.PostCommitVolumes,
+	}
+}
+
+type account string
+
+var _ driver.Valuer = account("")
+
+func (m1 account) Value() (driver.Value, error) {
+	ret, err := json.Marshal(strings.Split(string(m1), ":"))
```

Comment:
Why separate this to join it when scanning ? Why not directly put `X:Y` and just validate this before inserting ?

Reply:
My bad, it was already like this before